### PR TITLE
Create spec and cpp files for bonding table.

### PR DESCRIPTION
**Pull request by xavinux**
As it was asked here https://github.com/facebook/osquery/issues/1173 I've created the interface_bonds.table and interface_bonds.cpp files for the osquery bonding table. Compilation and test in an Ubuntu Server 14.04 64 bits were ok.

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by xavinux** [comment]
Also tested with four interfaces bond (eth0, eth1, eth2, eth3):
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
osquery> select * from interface_bonds;
+-------+---------+----------------------+---------------+------+------------+-------+--------+-----------+-----------+----------+----------+------------------------------+--------------+--------+
| bond  | primary | interfaces           | mode          | type | mii_status | speed | duplex | tx_errors | rx_errors | tx_bytes | rx_bytes | queue_id                     | active_slave | miimon |
+-------+---------+----------------------+---------------+------+------------+-------+--------+-----------+-----------+----------+----------+------------------------------+--------------+--------+
| bond0 | eth0    | eth0 eth1 eth2 eth3
 | active-backup | 1    | up         | 1000  | full   | 0         | 0         | 43153    | 113709   | eth0:0 eth1:0 eth2:0 eth3:0
 | eth0         | 100    |
+-------+---------+----------------------+---------------+------+------------+-------+--------+-----------+-----------+----------+----------+------------------------------+--------------+--------+
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
(Some columns go off the right side of the github-comment, you'll need to horizontal scroll)

File: osquery/tables/networking/linux/interface_bonds.cpp [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+namespace tables {
+
+QueryData genInterfaceBonds(QueryContext& context) {
+  Row r;
+  QueryData results;
+
+/* Read data from /sys/class/net/bonding_masters file and check for the bond name.*/
+FILE *read,*read2,*read3,*read4,*read5,*read6,*read7,*read8,*read9,*read10,*read11,*read12,*read13,*read14,*read15;
+
+      read = fopen("/sys/class/net/bonding_masters","r");
```
Comment (theopolis) [inline osquery/tables/networking/linux/interface_bonds.cpp]: Prefer to use the osquery `readFile` APIs. Check out https://github.com/facebook/osquery/blob/master/include/osquery/filesystem.h#L71

What you'll end up with is something similar to:
```cpp
std::string content;
auto status = readFile("/sys/class/net/bonding_masters", content);
r["bond"] = content;
```

**Comment by theopolis** [comment]
If there's a file in proc that corresponds to each column then you can create a mapping from column name to the respective path:
```cpp
const std::map<std::string, std::string> kBondingPaths = {
  {"bond", "/sys/class/net/bonding_masters"},
  {"next_column_name", "/next/file/path"},
};
```

**Comment by xavinux** [comment]
@theopolis thank you, I'll be updating this file with your suggestions.

**Pull request closed without merging**